OpenSim/Sandbox/MatlabScripts/c3d_reading.m

```diff
@@ -1,124 +1,214 @@
-marker_dep_metadata.setValueArrayForKey("labels", marker_labels);
-marker_table.setDependentsMetaData(marker_dep_metadata);
 
-double time_step{1.0 / acquisition->GetPointFrequency()};
-for f = 1:0 
-    
-    
-    rowVector = RowVector()
-    
-    
-    
-    for(auto it = marker_pts->Begin();
-        it != marker_pts->End();
-        ++it) {
-        auto pt = *it;
-        row[m++] = SimTK::Vec3{pt->GetValues().coeff(f, 0),
-                               pt->GetValues().coeff(f, 1),
-                               pt->GetValues().coeff(f, 2)};
-    }
+function table_rotated = rotateTableData(table, rotationMatrix)
 
-    
-marker_table.appendRow(0 + f * time_step, row);
+import org.opensim.modeling.*
 
+nLabels = table.getNumColumns();
+nRows = table.getNumRows();
 
-for iMarker = 0 : length(markerdata) - 1
+table_rotated = table();
+
+for it = 0 : nLabels - 1
 
     % get the column data for the marker
-    marker = markers.updDependentColumn(markerlabels(iMarker+1) );
+    table_column = table_rotated.updDependentColumnAtIndex( it   );
 
     % go through each element of the table column, rotate the Vec3, and write
     % back to the column.
-    for iRow = 0 : markers.getNumRows - 1
+    for iRow = 0 : nRows - 1
         % get Matlab vector marker position
-        vectorData = [marker.getElt(0,iRow).get(0)...
-                      marker.getElt(0,iRow).get(1)...
-                      marker.getElt(0,iRow).get(2)];
+        vectorData = [table_column.getElt(0,iRow).get(0)...
+                      table_column.getElt(0,iRow).get(1)...
+                      table_column.getElt(0,iRow).get(2)];
 
         % rotate the marker data
-        rotatedData = [rotationMatrix'*vectorData']';
+        rotatedData = [rotationMatrix*vectorData']';
 
         % Write the rotated data back to the Vec3TimesSeriesTable
-        Vec3(rotatedData(1),rotatedData(2),rotatedData(3))
-        
-        methodsview(new_table)
-                 
-          create an empty timesseriestable and use append every row Vec3
-    end
+        elem = Vec3(rotatedData(1),rotatedData(2),rotatedData(3));
+        % set the value of the element
+        table_column.set(iRow, elem);
+     end
+end
 end
 
-%% Print the rotated markers to trc file
-trcfileadapter = TRCFileAdapter();
-trcfileadapter.write(markers,'test_walking_rotated.trc');
-
-%% Rotate Force data
-
-for iForces = 0 : length(forcesdata) - 1
-
-    % get the column data for the marker
-    force = markers.updDependentColumn(forcelabels(iForce+1) );
-
-    % go through each element of the table column, rotate the Vec3, and write
-    % back to the column.
-    for iRow = 0 : forces.getNumRows - 1
-        % get Matlab vector marker position
-        vectorData = [force.getElt(0,iRow).get(0)...
-                      force.getElt(0,iRow).get(1)...
-                      force.getElt(0,iRow).get(2)];
+function rotations = rotateCoordinateSys(axis,rot1,axis2,rot2)
```

Comment:
You can combine the rotations into a single `rotationMatrix`, which this method would provide. You could then return a `SimTK::Rotation` and use its convenience constructors to construct it for various sequences including space or body fixed. Since in L158 you apply each axis rotation one at a time, I think that is effectively space fixed. For example R_bf = *RxRyRz* would be body-fixed but you apply them in order separately  *(Rz(Ry(Rx(D))) = (RzRyRx)D = (R_bf')D* which is space-fixed.